app/presenters/speedy_af/proxy/master_file.rb

```diff
@@ -22,7 +22,9 @@ def encoder_class
   end
 
   def find_encoder_class(klass_name)
-    ActiveEncode::Base.descendants.find { |c| c.name == klass_name }
+    return nil if klass_name.blank? || !Object.const_defined?(klass_name)
+    klass = Object.const_get(klass_name)
+    return klass if klass.ancestors.include?(ActiveEncode::Base)
   end
```

Comment:
There's a lot of duplicated code (not just this method) between this proxy and the model, is there a way to dry it out using mixin or something?

Reply:
Sure, this is probably a good time to clean it up.

Reply:
I looked but couldn't come up with any good alternative to this duplication right now.  I think the proper solution is the separation of the UI from the model by true presenters powered only by solr documents instead of speedy_af proxies powered by both solr and fedora, but that is considerable more work to go through the UI and make sure that we are using the presenter only.  It is a larger refactor that probably shouldn't hold up this PR.  Maybe there could be changes made to speedy_af that could move us in that direction without having to do quite as large a refactor?  But this effort might not really be worth it given the eventual move to avalon-bundle.